Hazel/Udp/UnityUdpClientConnection.cs

```diff
@@ -195,12 +193,11 @@ public override void ConnectAsync(byte[] bytes = null)
                 throw new HazelException("A SocketException occurred while initiating a receive operation.", e);
             }
 
-            this.InitializeKeepAliveTimer();
-
             // Write bytes to the server to tell it hi (and to punch a hole in our NAT, if present)
             // When acknowledged set the state to connected
             SendHello(bytes, () =>
             {
+                this.InitializeKeepAliveTimer();
```

Comment:
Nope, it's just a single line change that could happen in any PR. I would say that it pairs nicely with the other one though since it falls in a similar vein of "Handshake hasn't happened, much less the hello packet, why are we doing KeepAlive pings?"

Reply:
Got it and yeah, makes sense!